src/compas_timber/connections/t_butt.py

```diff
@@ -20,25 +21,54 @@ class TButtJoint(ButtJoint):
         The main beam to be joined.
     cross_beam : :class:`~compas_timber.parts.Beam`
         The cross beam to be joined.
+    mill_depth : float
+        The depth of the pocket to be milled in the cross beam.
 
     Attributes
     ----------
     main_beam : :class:`~compas_timber.parts.Beam`
         The main beam to be joined.
     cross_beam : :class:`~compas_timber.parts.Beam`
         The cross beam to be joined.
+    mill_depth : float
+        The depth of the pocket to be milled in the cross beam.
 
     """
 
     SUPPORTED_TOPOLOGY = JointTopology.TOPO_T
 
-    def __init__(self, main_beam=None, cross_beam=None, mill_depth=0, birdsmouth=False, **kwargs):
-        super(TButtJoint, self).__init__(main_beam, cross_beam, mill_depth, birdsmouth, **kwargs)
-
-    def restore_beams_from_keys(self, model):
-        """After de-serialization, restores references to the main and cross beams saved in the model."""
-        self.main_beam = model.element_by_guid(self.main_beam_guid)
-        self.cross_beam = model.element_by_guid(self.cross_beam_guid)
+    @property
+    def __data__(self):
+        data = super(TButtJoint, self).__data__
+        data["main_beam"] = self.main_beam
```

Comment:
commented on the new pr also. beams guids were used here to prevent serializing the same beam multiple times.
instead, we beam is serialized just once, as part of the model, then on de-serialization the instances are references back here by `restore_beams_from_keys`.